### DISCUSSION TITLE
import without install

### DISCUSSION THREAD

**Discussion by GHuserE** [community/community#123195] [Programming Help] [answered]
### Body

When writing a Python pck, how can you import it in some other location? Putting it on pypi? The code is not yet functional.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by CopperEagle** [comment, marked as answer]
Hey there @GHuserE 

Pip doesn't just use repository names, it can also work with local folders. So if you are developing, then you may want to use

```
pip3 install -e path/to/project
```

Where `-e` allows you to keep editing it. If you omit it you need to reinstall whenever you make changes. Note that this does not work out of the box, you need to add some setup information. The easy way to do this is by adding a `setup.py` file with the code. Here is a basic template:

```py
from setuptools import setup, find_packages


setup(
    name="packagename",
    version="1.0",
    description='Package description',
    author='YourGithubName',
    license='MIT license',
    packages=find_packages(),
)

```

Again, as with your last question, use this in combination with **virtual environments**. Much easier to manage packages *and development of such* that way.

**Reply by GHuserE** [reply]
thx
